Quick/Example.swift

```diff
@@ -71,6 +71,8 @@ private var numberOfExamplesRun = 0
             after(exampleMetadata: exampleMetadata)
         }
         world.exampleHooks.executeAfters(exampleMetadata)
+        
+        world.definitions.clearMemoizedValues()
```

Comment:
So, I'm leaning towards the opinion that while this shouldn't be included in Quick itself, there's no reason Quick can't provide hooks that would allow you to release a library that made it possible for people to use `define` within their specs.

Perhaps Quick (QuickCore as of #287) can provide a hook here that would allow you to execute code. You could define a `Definitions` singleton and clear its memoized values here.